runtime/src/main/java/com/flipkart/flux/deploymentunit/ExecutableRegistryPopulator.java

```diff
@@ -0,0 +1,146 @@
+                //for every task method found in the deployment unit create an executable and keep it in executable registry
+                for(Method method : taskMethods) {
+                    String taskIdentifier = new MethodId(method).toString();
+                    Annotation taskAnnotation = method.getAnnotationsByType(TaskClass)[0];
+                    Class<? extends Annotation> annotationType = taskAnnotation.annotationType();
+                    long timeout = 1000l; //default timeout
+                    for (Method annotationMethod : annotationType.getDeclaredMethods()) {
+                        Object value = annotationMethod.invoke(taskAnnotation, (Object[])null);
+                        if(annotationMethod.getName().equals("timeout")) { //todo: find a way get Task.timeout() name
+                            timeout = (Long) value;
+                        }
+                    }
+
+                    Object singletonMethodOwner = getClassInstanceMethod.invoke(injectorClassInstance, method.getDeclaringClass());
+                    executableRegistry.registerTask(taskIdentifier, new ExecutableImpl(singletonMethodOwner, method, timeout, classLoader));
+
+                    //add the router name to deployment unit routers
+                    routerNames.add(method.getDeclaringClass().getName() + "_" + method.getName()); //convention: task router name is classFQN_taskMethodName
+                }
+            } catch (Exception e) {
+                LOGGER.error("Unable to populate Executable Registry for deployment unit: "+deploymentUnitName+". Exception: "+e.getMessage());
+                throw new FluxError(FluxError.ErrorType.runtime, "Unable to populate Executable Registry for deployment unit: "+deploymentUnitName, e);
+            }
+        }
+
+    }
+
+    /**
+     * Loads {@Link ClassLoaderInjector} class into given deployment unit's class loader and returns it.
+     * @param deploymentUnitClassLoader
+     * @return ClassLoaderInjector class
+     */
+    private Class loadClassLoaderInjector(ClassLoader deploymentUnitClassLoader) {
+        try {
+            //Convert the class into bytes
+            byte[] classBytes = IOUtils.toByteArray(this.getClass().getResourceAsStream("/com/flipkart/flux/deploymentunit/ClassLoaderInjector.class"));
+
+            Class classLoaderInjectorClass;
+            Class classLoaderClass = Class.forName("java.lang.ClassLoader");
+            java.lang.reflect.Method method = classLoaderClass.getDeclaredMethod("defineClass", String.class, byte[].class, int.class, int.class);
+
+            // protected method invocation
+            method.setAccessible(true);
+            try {
+                Object[] args = new Object[]{ClassLoaderInjector.class.getCanonicalName(), classBytes, 0, classBytes.length};
+                classLoaderInjectorClass = (Class) method.invoke(deploymentUnitClassLoader, args);
```

Comment:
defineClass in URLClassLoader internally calls ClassLoader's method only and it's private. So I have to fetch it to use. It accepts params String className and sun.misc.Resource, I played with it by creating a Resource from these params, but it didn't workout. If I invoke defineClass directly from ClassLoader it uses default protection domain which has only read permissions on the resource also certs as null. It works well for our purpose.